sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClient.cs

```diff
@@ -130,19 +197,27 @@ public virtual async Task<Response<CommunicationUserIdentifier>> CreateUserAsync
             }
         }
 
+        /// <summary>Asynchronously creates a new <see cref="CommunicationUserIdentifier"/>.</summary>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        public virtual async Task<Response<CommunicationUserIdentifier>> CreateUserAsync(CancellationToken cancellationToken = default)
+        {
+            return await CreateUserAsync(default, cancellationToken).ConfigureAwait(false);
+        }
+
         /// <summary>Creates a new <see cref="CommunicationUserIdentifier"/>.</summary>
+        /// <param name="externalId">External Id to be associated with the user.</param>
         /// <param name="scopes">List of <see cref="CommunicationTokenScope"/> scopes for the token.</param>
         /// <param name="tokenExpiresIn">Custom validity period of the token within [1,24] hours range.</param>
         /// <param name="cancellationToken">The cancellation token to use.</param>
-        public virtual Response<CommunicationUserIdentifierAndToken> CreateUserAndToken(IEnumerable<CommunicationTokenScope> scopes, TimeSpan tokenExpiresIn, CancellationToken cancellationToken = default)
+        public virtual Response<CommunicationUserIdentifierAndToken> CreateUserAndToken(string externalId, IEnumerable<CommunicationTokenScope> scopes, TimeSpan tokenExpiresIn, CancellationToken cancellationToken = default)
```

Comment:
As we have already three optional parameters maybe we can have here options pattern for the parameters?https://azure.github.io/azure-sdk/dotnet_introduction.html
<img width="790" alt="image" src="https://github.com/user-attachments/assets/a62f6744-0308-4a0c-b786-32209e9ac069" />

Reply:
In the end we chose to go with the current approach